modules/mcagar/docs/bdd.md

```diff
@@ -0,0 +1,224 @@
+
+#### Example: Clicking a toolbar button and waiting for a dialog
+
+```js
+it('should open a dialog when clicking a toolbar button', async () => {
+  const editor = hook.editor();
+  TinyUiActions.clickOnToolbar(editor, 'button[aria-label="My Button"]');
+  await TinyUiActions.pWaitForDialog(editor);
+  TinyUiActions.closeDialog(editor);
+});
+```
+
+## TinySelections
+
+The `TinySelections` module provides helper functions to set the selection inside the editor.
+
+This module relies on defining an "element path" to specify the element to be selected. The element path is a number array of child node indexes starting from the root element. As an example, take the following table. The path to the `Cell 6` text node would be `[ 0, 1, 2, 0 ]`, or to explain using the element names `[ tbody, 2nd tr, 3rd td, text ]`.
+
+```html
+<table>
+  <tbody>
+    <tr>
+      <td>Cell 1</td>
+      <td>Cell 2</td>
+      <td>Cell 3</td>
+    </tr>
+    <tr>
+      <td>Cell 4</td>
+      <td>Cell 5</td>
+      <td>Cell 6</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+The following functions are available:
+- `setCursor(editor: Editor, path: number[], offset: number)`
+  - Sets a collapsed selection in the editor using the specified element path and offset.
+- `setSelection(editor: Editor, startPath: number[], startOffset: number, endPath: number[], endOffset: number)`
+  - Sets a selection that extends across multiple characters or elements in the editor by using the specified element paths and offsets.
+- `setSelectionFrom(editor: Editor, spec: CursorSpec | RangeSpec)`
+  - Sets the selection in the editor using a `Cursors` specification from the `Agar` library.
+- `select(editor: Editor, selector: string, path: number[])`
+  - Select an element inside the editor, such as a `img` element using a selector and element path.
+
+### Example: Setting the selection inside a paragraph
+
+```js
+it('should set the selection inside a paragraph', () => {
+  const editor = hook.editor();
+  editor.setContent('<p>Content</p>');
+  TinySelections.setSelection(editor, [ 0 ], 3, [ 0 ], 7); // Selects `tent` in the paragraph
+});
+```
+
+### Example: Selecting an image
+
+```js
+it('should set the selection around an image', () => {
+  const editor = hook.editor();
+  editor.setContent('<p><img src="..."></p>');
+  TinySelections.select(editor, 'img', []);
+});
+```
+
+## TinyAssertions
+
+The `TinyAssertions` module provides helper functions to assert the state of the editor.
+
+### Content assertions
+
+The following functions are available to assert the editor content:
+
+- `assertContent(editor: Editor, expectedContent: string)`
+  - Assert the content of the editor matches the expected content. The content is retrieved using `editor.getContent()`.
+- `assertRawContent(editor: Editor, expectedContent: string)`
+  - Assert the raw content of the editor. This is the HTML directly pulled from the editor body using `editor.getBody().innerHTML`.
```

Comment:
It might be worth adding a use case for `assertRawContent`, like when testing internal markup?

Reply:
```suggestion
  - Assert the raw content of the editor. This is the HTML directly pulled from the editor body using `editor.getBody().innerHTML`. This can be useful to assert content that only appears while editing, such as bogus br elements.
```